Added Longest Word in Every Language Article (#380)
TheRenegadeCoder/sample-programs-website: projects/longest-word-in-string/index.md

---

> title: Find length of the longest word in Every Language

Reviewer: ```suggestion
title: Longest Word in Every Language
```

Again, double check the path.

layout: default
date: 2020-10-25
last-modified: 2020-10-25
featured-image:

> tags: [longest-word-in-string, javascript]

Reviewer: ```suggestion
tags: [longest-word, javascript]
```

authors:
    - barhouum7
---

In this article, we'll outline the longest word in a string project.

> 

Reviewer: There should be a description section here that describes the problem in more detail with an example. I know we forgot it in your other documentation. I just wrote the following documentation as an example: https://sample-programs.therenegadecoder.com/projects/maximum-array-rotation/

Author: Good suggestion! I'll work on this.

## Requirements

The purpose of this program is to return the length of the longest word in the provided sentence as follows:

> ```javascript

Reviewer: ```suggestion
```shell
```

> longest-word-in-string.lang "Google do a barrel roll"

Reviewer: ```suggestion
$ ./longest-word-in-string.lang "Google do a barrel roll"
```

Author: Why do you want to make this, like we want to run a bash shell script?

Reviewer: This is just an example of how the file might be executed. It's not universal, but it shows how the program takes input from the command line.

Author: So you have to remove `./` if it is just an example of how a file of any language might be executed. Because `./` may refer of how executing a bash shell script.

> Output: "6"

Reviewer: ```suggestion
$ "6"
```

```

The result should be a number.

## Testing

The following table contains various test cases that you can use to verify the 
correctness of your solution:

| Description | Input | Output |
|--------------|-------|--------|
| No Input | | "Usage: please provide a string" |
| Empty Input | "" | "Usage: please provide a string" |

> | Simple Input | "May the force be with you" | "5" |

Reviewer: ```suggestion
| Sample Input: Many Words | "May the force be with you" | "5" |
```

> | Simple Input | "May the force be with you" | "5" |

Reviewer: This looks good! Perhaps we could add at least one more test case for a single word. Also, just like your other documentation, we need to define what a **word** is.

Author: Please check the last changes

## Articles

> {% include article_list.md collection=site.categories.longest-word-in-string %}

Reviewer: ```suggestion
{% include article_list.md collection=site.categories.longest-word %}
```
